server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java

```diff
@@ -122,6 +122,8 @@ protected NodesReloadSecureSettingsResponse.NodeResponse nodeOperation(
             keystore.decrypt(request.hasPassword() ? request.getSecureSettingsPassword().getChars() : new char[0]);
             // add the keystore to the original node settings object
             final Settings settingsWithKeystore = Settings.builder().put(environment.settings(), false).setSecureSettings(keystore).build();
+            clusterService.getClusterSettings().validate(settingsWithKeystore, true);
```

Comment:
Nice, this turned out quite simple! Unrelated to the test fix, but there's some docs we should update in a follow up:

- (Optional) it could be good to mention in a callout here that _all_ secure settings will be validated on a reload call, even static ones: https://www.elastic.co/guide/en/elasticsearch/reference/current/secure-settings.html#reloadable-secure-settings
- In cloud docs, we have a callout [here](https://www.elastic.co/guide/en/cloud/current/ec-configuring-keystore.html) stating that "the keystore has no validation to block unsupported settings. Adding unsupported settings to the keystore causes Elasticsearch to fail to start." we should update that to reflect the fact that we _do_ run validation on reload now, and let the cloud team know just as a heads up of this change

Reply:
Ah good catch! Will do a follow up.